code/lib/instrumenter/src/instrumenter.ts

```diff
@@ -412,12 +412,14 @@ export class Instrumenter {
     const { callRefsByResult, renderPhase } = this.getState(call.storyId);
 
     // Map complex values to a JSON-serializable representation.
-    const serializeValues = (value: any): any => {
+    // We use a depth, to avoid infinite recursion of self referencing values.
+    const serializeValues = (value: any, depth = 0): any => {
       if (callRefsByResult.has(value)) {
         return callRefsByResult.get(value);
       }
       if (value instanceof Array) {
-        return value.map(serializeValues);
+        if (depth > 10) return [];
```

Comment:
Please add a test case to cover this scenario

Reply:
done